src/Npgsql/TypeHandlers/LQueryHandler.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Npgsql.BackendMessages;
+using Npgsql.PostgresTypes;
+using Npgsql.TypeHandling;
+using Npgsql.TypeMapping;
+using NpgsqlTypes;
+
+namespace Npgsql.TypeHandlers
+{
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+
+    [TypeMapping("lquery", NpgsqlDbType.LQuery)]
+    class LQueryHandlerFactory : NpgsqlTypeHandlerFactory<string>
+    {
+        public override NpgsqlTypeHandler<string> Create(PostgresType postgresType, NpgsqlConnection conn)
+            => new LQueryHandler(postgresType, conn);
+    }
+
+    /// <summary>
+    /// LQuery binary encoding is a simple UTF8 string, but prepended with a version number.
+    /// </summary>
+    public class LQueryHandler : TextHandler
```

Comment:
Since this is largely duplicated with other handlers, we could simply make it generic, i.e. VersionPrepdnedTextHandler<T> which returns T and does the version logic. Not mandatory for now obviously.